src/include/miopen/env.hpp

```diff
@@ -121,22 +121,7 @@ struct EnvVar
         if(vp != nullptr) // a value was provided
         {
             is_default = false;
-            if constexpr(std::is_same_v<T, bool>)
-            {
-                value = ParseEnvVal<bool>::go(vp);
-            }
-            else if constexpr(std::is_same_v<T, uint64_t>)
-            {
-                value = ParseEnvVal<uint64_t>::go(vp);
-            }
-            else if constexpr(std::is_same_v<T, std::string>)
-            {
-                value = ParseEnvVal<std::string>::go(vp);
-            }
-            else
-            {
-                value = ParseEnvVal<T>::go(vp); // should cause compile error
-            }
+            value      = ParseEnvVal<T>::go(vp);
```

Comment:
Good catch @JehandadKhan . This one's on me as I gave Chris the starter code and didn't see it.